### ISSUE TITLE
twisted.internet.iocpreactor.iocpsupport.fillinetaddr should take an address family parameter and support AF_INET6

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5116]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| PenguinOfDoom reported|
|-|-|
|Trac ID|trac#5116|
|Type|enhancement|
|Created|2011-05-20 18:57:06Z|
|Branch|https://github.com/twisted/twisted/tree/iocp-connect-ipv6-5116|

iocpreactor's ConnectEx wrapper does not currently support AF_INET6 sockets and addresses

<details><summary>Searchable metadata</summary>

```
trac-id__5116 5116
type__enhancement enhancement
reporter__PenguinOfDoom PenguinOfDoom
priority__normal normal
milestone__ 
branch__branches_iocp_connect_ipv6_5116 branches/iocp-connect-ipv6-5116
branch_author__pahan pahan
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1305917826000000 1305917826000000
changetime__1474197555542286 1474197555542286
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|pahan commented|
|-|-|

(In [31910]) Branching to 'iocp-connect-ipv6-5116'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

This is fixed in the same branch as [#5085](https://github.com/twisted/twisted/issues/5085).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph removed owner<br>@glyph set status to `new`|
|-|-|

Putting this up for review because the reviewer of [#5085](https://github.com/twisted/twisted/issues/5085) definitely deserves some bonus points.  (And, if they are so inclined, they may review this portion separately.)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

(In [34020]) Merge ipv6-connectTCP-5085-2.

Author: acapnotic, glyph, chjurk, ragzilla

Reviewer: exarkun

Allow Twisted to connect to IPv6 addresses, using all currently supported reactors.

IPv6 clients are supported by passing an IPv6 address literal to `connectTCP`.  Hostname resolution is the responsibility of the caller.

This change also significantly improves the internal maintenance documentation coverage for `twisted.internet.tcp` and refactors more client code to be shared between the IOCP reactor and all the UNIX reactors.

Fixes: [#5085](https://github.com/twisted/twisted/issues/5085)
Fixes: [#5116](https://github.com/twisted/twisted/issues/5116)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
